tools/license/get_dep_licenses.go

```diff
@@ -45,7 +47,85 @@ const (
 	maxLevelsToLicense = 7
 )
 
+type licenseType int
+
+const (
+	// licenseTypeApproved is definitely ok to use and modify.
+	licenseTypeApproved licenseType = iota
+	// licenseTypeReciprocal can be used but not modified.
+	licenseTypeReciprocal
+	// licenseTypeRestricted
+	licenseTypeRestricted
+)
+
 var (
+	// strToLicenseType are code that's definitely ok to use and modify.
+	strToLicenseType = map[string]licenseType{
```

Comment:
Thanks for making this change. =)

Should/can this be called something like licenseToType?  And I also wonder if the following will be more readable if the constants are just Approved, Restricted, etc.

Reply:
I changed it licenseStrToType because it really is a string, not a license that we're mapping. 
Regarding the enum labels, in Go it's normal practice to have longer, more qualified labels because names are all package scoped. So it would be "approved" not "Approved" (it's private) and then you couldn't create a var called approved because it clashes. All this crap gets beat into when you go through Go readability... is the bool map starting to look better now? :-D